numpy/lib/function_base.py

```diff
@@ -4326,6 +4327,69 @@ def meshgrid(*xi, **kwargs):
             return np.broadcast_arrays(*output)
 
 
+def meshgridify(*xi, **kwargs):
```

Comment:
First off, I don't really have an opinion about adding this. New functions should also go to the mailing list for more general discussion on inclusion and the API. There may also be other packages this fits better.

Not sure I like the name much, I think it may be possible to find something better.